skule_vote/backend/serializers.py

```diff
@@ -41,3 +43,54 @@ class Meta:
             "start_time",
             "end_time",
         )
+
+
+class BallotSerializer(serializers.Serializer):
+    electionId = serializers.CharField(max_length=200, allow_blank=False)
+    # rank -> candidate_id
+    ranking = serializers.DictField(allow_empty=True)
+
+    def validate(self, data):
+        """
+        Ensure election exists and candidates belong to the election.
+        """
+        if Election.objects.filter(id=data["electionId"]).count() != 1:
+            raise serializers.ValidationError(
+                "The election you are trying to vote in does not exist"
+            )
```

Comment:
This piece of code never gets executed since the same error is caught in the `views.py`